scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,423 @@
+            {
+                return AudioConfig.FromWavFileInput(inputFileValue);
+            }
+            else
+            {
+                return AudioConfig.FromDefaultMicrophoneInput();
+            }
+        }
+
+        static SpeechConfig SpeechConfigFromUserConfig(UserConfig userConfig)
+        {
+            SpeechConfig speechConfig;
+            if (userConfig.languageIDLanguages is not null)
+            {
+                speechConfig = SpeechConfig.FromEndpoint(new Uri(V2EndpointFromRegion(userConfig.region)), userConfig.subscriptionKey);
+            }
+            else
+            {
+                speechConfig = SpeechConfig.FromSubscription(userConfig.subscriptionKey, userConfig.region);
+            }
+
+            if (userConfig.profanityFilterRemoveEnabled)
+            {
+                speechConfig.SetProfanity(ProfanityOption.Removed);
+            }
+            else if (userConfig.profanityFilterMaskEnabled)
+            {
+                speechConfig.SetProfanity(ProfanityOption.Masked);
+            }
+            
+            if (userConfig.stablePartialResultThreshold is string stablePartialResultThresholdValue)
+            {
+                speechConfig.SetProperty(PropertyId.SpeechServiceResponse_StablePartialResultThreshold, stablePartialResultThresholdValue);
+            }
+            
+            if (userConfig.trueTextEnabled)
+            {
+                speechConfig.SetProperty(PropertyId.SpeechServiceResponse_PostProcessingOption, "TrueText");
+            }
+            
+            return speechConfig;
+        }
+
+        static SpeechRecognizer SpeechRecognizerFromSpeechConfig(SpeechConfig speechConfig, AudioConfig audioConfig, UserConfig userConfig)
+        {
+            SpeechRecognizer speechRecognizer;
+            if (userConfig.languageIDLanguages is string[] languageIDLanguagesValue)
+            {
+                AutoDetectSourceLanguageConfig detectLanguageConfig = AutoDetectSourceLanguageConfig.FromLanguages(languageIDLanguagesValue);
+                speechRecognizer = new SpeechRecognizer(speechConfig, detectLanguageConfig, audioConfig);
+            }
+            else
+            {
+                speechRecognizer = new SpeechRecognizer(speechConfig, audioConfig);
+            }
+            
+            if (userConfig.phraseList is string phraseListValue)
+            {
+                PhraseListGrammar grammar = PhraseListGrammar.FromRecognizer(speechRecognizer);
+                grammar.AddPhrase(phraseListValue);
+            }
+            
+            return speechRecognizer;
+        }
+
+        static SpeechRecognizer SpeechRecognizerFromUserConfig(UserConfig userConfig)
+        {
+            AudioConfig audioConfig = AudioConfigFromUserConfig(userConfig);
+            SpeechConfig speechConfig = SpeechConfigFromUserConfig(userConfig);
+            return SpeechRecognizerFromSpeechConfig(speechConfig, audioConfig, userConfig);
```

Comment:
@v-jaswel - "FromSpeechConfig" does not quite capture how a SpeechRecognizer should be constructed. Also not clear why do we need an extra helper method? I suggest replace the current `return` line with:

```csharp
            SpeechRecognizer speechRecognizer;
            if (userConfig.languageIDLanguages is string[] languageIDLanguagesValue)
            {
                AutoDetectSourceLanguageConfig autoDetectSourceLanguageConfig = AutoDetectSourceLanguageConfig.FromLanguages(languageIDLanguagesValue);
                speechRecognizer = new SpeechRecognizer(speechConfig, autoDetectSourceLanguageConfig, audioConfig);
            }
            else
            {
                speechRecognizer = new SpeechRecognizer(speechConfig, audioConfig);
            }

            if (userConfig.phraseList is string phraseListValue)
            {
                PhraseListGrammar grammar = PhraseListGrammar.FromRecognizer(speechRecognizer);
                grammar.AddPhrase(phraseListValue);
            }

            return speechRecognizer;
```